src/platform/user/authentication/utilities.js

```diff
@@ -75,12 +68,25 @@ function redirectWithGAClientId(redirectUrl) {
   }
 }
 
+export function standaloneRedirect() {
+  const searchParams = new URLSearchParams(window.location.search);
+  const application = searchParams.get('application');
+  const to = searchParams.get('to');
+  let url = externalRedirects[application] || null;
+
+  if (url && to) {
+    url = url.replace(new RegExp('/$'), '');
+    url += `/${to.replace(new RegExp('^/'), '').replace('\r\n', '')}`;
```

Comment:
Could `trim()` be used instead for the CRLF?

Reply:
Maybe more readable as:

```
let url;
const hostname = externalRedirects[application];

if (hostname) {
  const pathname = to.startsWith('/') ? to : `/${to}`;
  url = `${hostname}/${pathname};
  url.trim();
}
```

Reply:
The idea is that a malicious CRLF could appear anywhere, not just at the end of the URL.

Reply:
Makes sense. Would rather see the validation on the backend, but in lieu of that, I would amend my suggestion to replace like the original:

```
url = `${baseUrl}${pathname}`.replace('\r\n', ''); // Prevent CRLF injection.
```